hasktorch/src/Torch/Typed/Tensor.hs

```diff
@@ -120,9 +126,11 @@ instance (KnownNat n) => KnownDevice '( 'D.CPU, n) where
 instance (KnownNat n) => KnownDevice '( 'D.CUDA, n) where
   deviceVal = D.Device D.CUDA (natValInt16 @n)
 
-data Tensor (device :: (D.DeviceType, Nat)) (dtype :: D.DType) (shape :: [Nat]) where
+data Tensor (device :: (D.DeviceType, Nat)) (dtype :: D.DType) (shape :: [Type->Type]) where
   UnsafeMkTensor :: forall device dtype shape. {toDynamic :: D.Tensor} -> Tensor device dtype shape
 
+type Tensor' device dtype shape = Tensor device dtype (FromNats shape)
+
```

Comment:
This PR defines old tensor(Tensor') using [Nat] as type synonym.
It occurs an illegal type synonym family application error in a LSTM implementation .